kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestAvroWriter.java

```diff
@@ -44,9 +44,9 @@
                 String.valueOf(32 * 1024 * 1024)) // 32 MB
             .schema(schema)
             .format("avro")
-            .build());
+            .build(), TEST_SCHEMA);
```

Comment:
This helper method shouldn't directly access TEST_SCHEMA. If you want to be able to use it to test this case, then lets use separate datasetSchema and writerSchema. Also, I think you have it backwards. This uses the passed schema for the dataset's schema. That's probably why it isn't catching the incompatibility.